### PR TITLE
Introduce `Libplanet.Crypto` & `Libplanet.Types`

### PR DESCRIPTION
**Pull request by OnedgeLee**
Extract `Libplanet.Crypto` and `Libplanet.Types` from `Libplanet.Common`.

### CODE REVIEW

File: Libplanet/Libplanet.csproj [hunk 1]
```diff
@@ -62,15 +62,6 @@ https://docs.libplanet.io/</Description>
...
   <ItemGroup>
-    <PackageReference Include="Bencodex" Version="0.11.0" />
-    <PackageReference Include="Bencodex.Json" Version="0.11.0" />
-    <PackageReference Include="BouncyCastle.Cryptography" Version="2.0.0" />
-    <PackageReference Include="Caching.dll" Version="1.4.0.1" />
-    <PackageReference Include="Destructurama.Attributed" Version="2.0.0" />
-    <PackageReference Include="ImmutableTrie" Version="1.0.0-alpha" />
-    <PackageReference Include="LiteDB" Version="4.1.4" />
-    <PackageReference Include="Microsoft.Bcl.HashCode" Version="1.1.1" />
-    <PackageReference Include="Planetarium.LruCacheNet" Version="1.2.0" />
```
Comment (greymistcube) [inline Libplanet/Libplanet.csproj:73]: Not sure entirely removing dependency is great. This relies on knowing each subproject's dependency. 🤔

Comment (OnedgeLee) [reply Libplanet/Libplanet.csproj:73]: Not sure for this, too. hmm...

File: Libplanet.sln [hunk 2]
```diff
@@ -49,7 +49,11 @@ Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Libplanet.Common", "Libplan
 EndProject
 Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Libplanet.Store", "Libplanet.Store\Libplanet.Store.csproj", "{2BC01C4C-288B-4768-BAD0-9BC441D82505}"
 EndProject
-Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "Libplanet.Action", "Libplanet.Action\Libplanet.Action.csproj", "{DDD598B2-005F-4CB2-B26B-A40CC0F9DE8E}"
+Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Libplanet.Action", "Libplanet.Action\Libplanet.Action.csproj", "{DDD598B2-005F-4CB2-B26B-A40CC0F9DE8E}"
+EndProject
+Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "Libplanet.Crypto", "Libplanet.Crypto\Libplanet.Crypto.csproj", "{A94DDE86-B103-48DE-A8D3-B9C80ABDA887}"
+EndProject
+Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "Libplanet.Types", "Libplanet.Types\Libplanet.Types.csproj", "{65E7ECA8-34EC-450D-B6AE-245C6717AB20}"
```
[the remark this answers is not in the document]
Comment (OnedgeLee) [reply Libplanet.sln:56]: Found this issue. https://github.com/dotnet/project-system/issues/1821

File: Libplanet/Libplanet.csproj (hunk 1 above)
Comment (OnedgeLee) [reply Libplanet/Libplanet.csproj]: My preference is 50% and 50%.

Comment (greymistcube) [reply Libplanet/Libplanet.csproj]: Well, I haven't thought about it too deeply, but if the project is well organized either
- Having different versions shouldn't matter
- Shouldn't reference the "lower" packages in the first place
If we are having trouble, I'd say that's generally indicative of some code smell.

File: .github/bin/constants.sh [hunk 3]
```diff
@@ -4,9 +4,11 @@
...
   "Libplanet"
-  "Libplanet.Action"
   "Libplanet.Common"
+  "Libplanet.Crypto"
+  "Libplanet.Types"
   "Libplanet.Store"
+  "Libplanet.Action"
   "Libplanet.Stun"
   "Libplanet.Net"
   "Libplanet.Crypto.Secp256k1"
```
Comment (riemannulus) [inline .github/bin/constants.sh:14]:
```suggestion
  "Libplanet.Action"
  "Libplanet.Common"
  "Libplanet.Crypto"
  "Libplanet.Crypto.Secp256k1"
  "Libplanet.Net"
  "Libplanet.Store"
  "Libplanet.Stun"
  "Libplanet.Types"
```

File: Libplanet/Libplanet.csproj [hunk 4]
```diff
@@ -62,15 +62,12 @@ https://docs.libplanet.io/</Description>
   </ItemGroup>
 
   <ItemGroup>
-    <PackageReference Include="Bencodex" Version="0.11.0" />
-    <PackageReference Include="Bencodex.Json" Version="0.11.0" />
+    <PackageReference Include="System.Collections.Immutable" Version="1.*" />
+    <PackageReference Include="System.Text.Json" Version="6.0.*" />
+    <PackageReference Include="Bencodex" Version="0.10.0" />
```
Comment (greymistcube) [inline Libplanet/Libplanet.csproj:67]: `Bencodex` downgrade detected. 😕

**Review by limebell — DISMISSED** [review]

**Pull request merged**